### DISCUSSION TITLE
pageLength not working as expected in angular datatable server side response

### DISCUSSION THREAD

**Discussion by srijanlama** [l-lin/angular-datatables#1647] [Q&A]
# :beetle: bug report

`pageLength = 2` not returning expected result on datatable 
```
  ngOnInit(): void {
    const that = this;
    this.dtOptions = {
      pagingType: 'full_numbers',
      pageLength: 2, // this should return datatable of page length 2 
      serverSide: true,
      processing: true,
      ajax: (dataTablesParameters: any, callback) => {
        that.http
          .post<DataTablesResponse>(
            environment.api_url + `job/manpower`,
            dataTablesParameters,
            {}
          )
          .subscribe((res: any) => {
            that.jobs = res.data;
            callback({
              recordsTotal: res.recordsTotal,
              data: [],
            });
          });
      },
      columns: [{ data: 'id' }, { data: 'title.en' }],
    };
  }
```
### server api response 

![api-data](https://user-images.githubusercontent.com/42295335/158942353-2471529a-4c3d-4d12-bf5b-069f90bdf311.PNG)

### request payload 

![payload](https://user-images.githubusercontent.com/42295335/158942421-76cc75ab-c3de-41c6-9e0f-56dbe1818f73.PNG)

datatables shows more than 2 records 

![SajhajobsFrontend](https://user-images.githubusercontent.com/42295335/158942719-662fbc87-2b24-4385-9bb2-7c555e066bc8.png)

- NodeJS version: 14.15.1
- Angular version: 13.1.0
- Angular CLI version: 13.1.0
- jQuery version: ^3.6.0
- DataTables version: 1.11.3
- angular-datatables version: ^13.0.1

any help is Appreciated 
Sri Zn

**Comment by shanmukhateja** [comment]
Hello,

Can you provide a StackBlitz link or GitHub repo to investigate this?

**Reply by srijanlama** [reply]
thanks for repley @shanmukhateja  , the api response coming from laravel backend  . i will try to replicate on stackbiz later
